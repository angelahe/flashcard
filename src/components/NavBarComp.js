import React from 'react';
import PropTypes from 'prop-types';
import decksBtn from '../img/buttons/decks_FFFFFF.png';
import decksOffBtn from '../img/buttons/decksOff_FFFFFF.png';
import listBtn from '../img/buttons/list_FFFFFF.png';
import listOffBtn from '../img/buttons/listOff_FFFFFF.png';
import chartBtn from '../img/buttons/chart_FFFFFF.png';
import chartOffBtn from '../img/buttons/chartOff_FFFFFF.png';
import settingsBtn from '../img/buttons/settings_FFFFFF.png';
import settingsOffBtn from '../img/buttons/settingsOff_FFFFFF.png';
import infoBtn from '../img/buttons/info_FFFFFF.png';
import infoOffBtn from '../img/buttons/infoOff_FFFFFF.png';
import '../styles/flashcard.css';

class NavBarComp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentFocus: 'list',
    };
  }

  handleDecksClicked = () => {
    const { onNavClick } = this.props;
    this.setState({ currentFocus: 'decks' });
    onNavClick('FlashcardDecks');
  };

  handleListClicked = () => {
    const { onNavClick } = this.props;
    this.setState({ currentFocus: 'list' });
    onNavClick('ManageDecks');
  };

  handleChartClicked = () => {
    const { onNavClick } = this.props;
    this.setState({ currentFocus: 'chart' });
    onNavClick('FlashcardChart');
  };

  handleSettingsClicked = () => {
    const { onNavClick } = this.props;
    this.setState({ currentFocus: 'settings' });
    onNavClick('FlashcardSettings');
  };

  handleInfoClicked = () => {
    const { onNavClick } = this.props;
    this.setState({ currentFocus: 'info' });
    onNavClick('FlashcardInfo');
  };

  render() {
    const {
      currentFocus,
    } = this.state;

    return (
      <div className="NavBarComp NavBar">
        <div className="LineContainer">
          { currentFocus === 'decks'
            ? (
              <button type="button" className="NavBarDecksOn NavBtn">
                <img className="NavImg" src={decksBtn} alt="Decks" />
              </button>
            )
            : (
              <button type="button" className="NavBarDecksOff NavBarNavBtn" onClick={this.handleDecksClicked}>
                <img className="NavImg" src={decksOffBtn} alt="Decks" />
              </button>
            )
          }
          { currentFocus === 'list'
            ? (
              <button type="button" className="NavBarListOn NavBtn">
                <img className="NavImg" src={listBtn} alt="List" />
              </button>
            )
            : (
              <button type="button" className="NavBarListOff NavBtn" onClick={this.handleListClicked}>
                <img className="NavImg" src={listOffBtn} alt="List" />
              </button>
            )
          }
          { currentFocus === 'chart'
            ? (
              <button type="button" className="NavBarChartOn NavBtn">
                <img className="NavImg" src={chartBtn} alt="Chart" />
              </button>
            )
            : (
              <button type="button" className="NavBarChartOff NavBtn" onClick={this.handleChartClicked}>
                <img className="NavImg" src={chartOffBtn} alt="Chart" />
              </button>
            )
          }
          { currentFocus === 'settings'
            ? (
              <button type="button" className="NavBarSettingsOn NavBtn">
                <img className="NavImg" src={settingsBtn} alt="Settings" />
              </button>
            )
            : (
              <button type="button" className="NavBarSettingsOff NavBtn" onClick={this.handleSettingsClicked}>
                <img className="NavImg" src={settingsOffBtn} alt="Settings" />
              </button>
            )
          }
          { currentFocus === 'info'
            ? (
              <button type="button" className="NavBarInfoOn NavBtn">
                <img className="NavImg" src={infoBtn} alt="Info" />
              </button>
            )
            : (
              <button type="button" className="NavBarInfoOff NavBtn" onClick={this.handleInfoClicked}>
                <img className="NavImg" src={infoOffBtn} alt="Info" />
              </button>
            )
          }
        </div>
      </div>
    );
  }
}

NavBarComp.defaultProps = {
  onNavClick: () => { },
};

NavBarComp.propTypes = {
  onNavClick: PropTypes.func,
};

export default NavBarComp;
