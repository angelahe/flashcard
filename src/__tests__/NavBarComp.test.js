import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import NavBarComp from '../components/NavBarComp';

it('renders without crashing', () => {
  shallow(<NavBarComp />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NavBarComp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('NavBarComp renders correctly', () => {
  const tree = renderer.create(<NavBarComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

// default to show ManageDecks is list
it('contains the container div NavBarComp on shallow render with decks focus as default', () => {
  const component = shallow(<NavBarComp />);
  expect(component.instance().state.currentFocus).toBe('list');
  expect(component.html()).toContain('NavBarComp');
  expect(component.html()).toContain('NavBarDecksOff');
  expect(component.html()).toContain('NavBarListOn');
  expect(component.html()).toContain('NavBarChartOff');
  expect(component.html()).toContain('NavBarSettingsOff');
  expect(component.html()).toContain('NavBarInfoOff');
});

it('contains NavBarComp and changes focus to NavBarDecks', (done) => {
  const handleNavClick = jest.fn(() => {});
  const component = shallow(<NavBarComp onNavClick={handleNavClick} />);
  const button = component.find('.NavBarDecksOff').first();
  button.simulate('click');

  setTimeout(() => {
    expect(component.instance().state.currentFocus).toBe('decks');
    expect(component.html()).toContain('NavBarComp');
    expect(component.html()).toContain('NavBarDecksOn');
    expect(component.html()).toContain('NavBarListOff');
    expect(component.html()).toContain('NavBarChartOff');
    expect(component.html()).toContain('NavBarSettingsOff');
    expect(component.html()).toContain('NavBarInfoOff');
    expect(handleNavClick.mock.calls.length).toBe(1);
    expect(handleNavClick.mock.calls[0][0]).toBe('FlashcardDecks');
    done();
  });
});

it('contains NavBarComp and changes focus to NavBarChart', (done) => {
  const handleNavClick = jest.fn(() => {});
  const component = shallow(<NavBarComp onNavClick={handleNavClick} />);
  const button = component.find('.NavBarChartOff').first();
  button.simulate('click');

  setTimeout(() => {
    expect(component.instance().state.currentFocus).toBe('chart');
    expect(component.html()).toContain('NavBarComp');
    expect(component.html()).toContain('NavBarDecksOff');
    expect(component.html()).toContain('NavBarListOff');
    expect(component.html()).toContain('NavBarChartOn');
    expect(component.html()).toContain('NavBarSettingsOff');
    expect(component.html()).toContain('NavBarInfoOff');
    expect(handleNavClick.mock.calls.length).toBe(1);
    expect(handleNavClick.mock.calls[0][0]).toBe('FlashcardChart');
    done();
  });
});

it('contains NavBarComp and changes focus to NavBarSettings', (done) => {
  const handleNavClick = jest.fn(() => {});
  const component = shallow(<NavBarComp onNavClick={handleNavClick} />);
  const button = component.find('.NavBarSettingsOff').first();
  button.simulate('click');

  setTimeout(() => {
    expect(component.instance().state.currentFocus).toBe('settings');
    expect(component.html()).toContain('NavBarComp');
    expect(component.html()).toContain('NavBarDecksOff');
    expect(component.html()).toContain('NavBarListOff');
    expect(component.html()).toContain('NavBarChartOff');
    expect(component.html()).toContain('NavBarSettingsOn');
    expect(component.html()).toContain('NavBarInfoOff');
    expect(handleNavClick.mock.calls.length).toBe(1);
    expect(handleNavClick.mock.calls[0][0]).toBe('FlashcardSettings');
    done();
  });
});

it('contains NavBarComp and changes focus to NavBarInfo', (done) => {
  const handleNavClick = jest.fn(() => {});
  const component = shallow(<NavBarComp onNavClick={handleNavClick} />);
  const button = component.find('.NavBarInfoOff').first();
  button.simulate('click');

  setTimeout(() => {
    expect(component.instance().state.currentFocus).toBe('info');
    expect(component.html()).toContain('NavBarComp');
    expect(component.html()).toContain('NavBarDecksOff');
    expect(component.html()).toContain('NavBarListOff');
    expect(component.html()).toContain('NavBarChartOff');
    expect(component.html()).toContain('NavBarSettingsOff');
    expect(component.html()).toContain('NavBarInfoOn');
    expect(handleNavClick.mock.calls.length).toBe(1);
    expect(handleNavClick.mock.calls[0][0]).toBe('FlashcardInfo');
    done();
  });
});

it('contains NavBarComp and changes focus to chart then NavBarDecks', (done) => {
  // first navigate away from default decks on
  const handleNavClick = jest.fn(() => {});
  const component = shallow(<NavBarComp onNavClick={handleNavClick} />);
  const button = component.find('.NavBarChartOff').first();
  button.simulate('click');
  const button2 = component.find('.NavBarListOff').first();
  button2.simulate('click');

  setTimeout(() => {
    expect(component.instance().state.currentFocus).toBe('list');
    expect(component.html()).toContain('NavBarComp');
    expect(component.html()).toContain('NavBarDecksOff');
    expect(component.html()).toContain('NavBarListOn');
    expect(component.html()).toContain('NavBarChartOff');
    expect(component.html()).toContain('NavBarSettingsOff');
    expect(component.html()).toContain('NavBarInfoOff');
    expect(handleNavClick.mock.calls.length).toBe(2);
    expect(handleNavClick.mock.calls[1][0]).toBe('ManageDecks');
    done();
  });
});
