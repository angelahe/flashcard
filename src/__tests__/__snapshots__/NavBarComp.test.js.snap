// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarComp renders correctly 1`] = `
<div
  className="NavBarComp NavBar"
>
  <div
    className="LineContainer"
  >
    <button
      className="NavBarDecksOff NavBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Decks"
        className="NavImg"
        src="decksOff_FFFFFF.png"
      />
    </button>
    <button
      className="NavBarListOn NavBtn"
      type="button"
    >
      <img
        alt="List"
        className="NavImg"
        src="list_FFFFFF.png"
      />
    </button>
    <button
      className="NavBarChartOff NavBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Chart"
        className="NavImg"
        src="chartOff_FFFFFF.png"
      />
    </button>
    <button
      className="NavBarSettingsOff NavBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Settings"
        className="NavImg"
        src="settingsOff_FFFFFF.png"
      />
    </button>
    <button
      className="NavBarInfoOff NavBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Info"
        className="NavImg"
        src="infoOff_FFFFFF.png"
      />
    </button>
  </div>
</div>
`;
